src/keeshare/ShareObserver.cpp

```diff
@@ -259,8 +259,12 @@ void ShareObserver::reinitialize()
     notifyAbout(success, warning, error);
 }
 
-void ShareObserver::notifyAbout(const QStringList& success, const QStringList& warning, const QStringList& error)
+void ShareObserver::notifyAbout(QStringList& success, const QStringList& warning, const QStringList& error)
```

Comment:
Don't change the signature of the function. Please add the const back in.

I think we need to evaluate what this function is doing. We only ever want to notify about one message type at a time. The signature should really be: ```void ShareObserver::notifyAbout(const QStringList& msgs, MessageType type)```. Where ```MessageType``` is an enum of ```SUCCESS, WARNING, ERROR```. Then you can clearly filter out the success messages.

Reply:
In my usage, I have multiple sync set. When all goes well, I see the 3 success messages. One day a co-worker renamed a shared folder without thinking, so next time I got 2 success and 1 error bundled in one red notification. The success messages confused me more than anything.

With my patch, success messages can be stripped away. So on a regular day, no notice shows, but on a error day it would be weird to see 2 success and 1 error.

I know where you are coming from though. It was a bit cavalier of me to just strip const. I'll tweak my implementation so I don't join the success list instead.

Reply:
Yah that is the better approach in case we add logging which would rely on the messages not being stripped.